float32[9] rotationMatrix
float32[3] translationVector